components/director/internal/systemfetcher/loader.go

```diff
@@ -341,3 +342,20 @@ func retrieveRegion(labels map[string]interface{}) (string, error) {
 	}
 	return regionValue, nil
 }
+
+func areAppTemplatesEqual(appTemplate *model.ApplicationTemplate, appTemplateInput model.ApplicationTemplateInput) bool {
+	if appTemplate == nil {
+		return false
+	}
+
+	isAppInputJSONEqual := reflect.DeepEqual(appTemplate.ApplicationInputJSON, appTemplateInput.ApplicationInputJSON)
```

Comment:
do you really need DeepEqual for strings?

Reply:
Yeah, you are right. Before this change, they were deepEqually checked and that's why I left it that way. But it seems that it is not necessary